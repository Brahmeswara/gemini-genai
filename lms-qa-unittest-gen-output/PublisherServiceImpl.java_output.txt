```java
package com.example.lws.services;

import com.example.lws.models.Publisher;
import com.example.lws.repository.PublisherRepository;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.mockito.junit.MockitoJUnitRunner;

import java.util.Optional;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.*;

@RunWith(MockitoJUnitRunner.class)
public class PublisherServiceImplTest {

    @InjectMocks
    private PublisherServiceImpl publisherService;

    @Mock
    private PublisherRepository publisherRepository;

    @Before
    public void setup() {
        MockitoAnnotations.initMocks(this);
    }

    @Test
    public void shouldSavePublisher() {
        Publisher publisher = Publisher.builder().name("Test Publisher").build();

        when(publisherRepository.save(any(Publisher.class))).thenReturn(publisher);

        Publisher savedPublisher = publisherService.savePublisher(publisher);

        verify(publisherRepository, times(1)).save(publisher);
        assert savedPublisher != null;
        assert savedPublisher.getId() != null;
    }

    @Test
    public void shouldUpdatePublisher() {
        Publisher publisher = Publisher.builder().id(1L).name("Test Publisher").build();

        when(publisherRepository.save(any(Publisher.class))).thenReturn(publisher);

        Publisher updatedPublisher = publisherService.updatePublisher(publisher);

        verify(publisherRepository, times(1)).save(publisher);
        assert updatedPublisher != null;
        assert updatedPublisher.getId() != null;
    }

    @Test
    public void shouldDeletePublisher() {
        Long publisherId = 1L;

        publisherService.deletePublisher(publisherId);

        verify(publisherRepository, times(1)).deleteById(publisherId);
    }

    @Test
    public void shouldGetPublisherById() {
        Long publisherId = 1L;
        Publisher publisher = Publisher.builder().id(publisherId).name("Test Publisher").build();

        when(publisherRepository.findById(any(Long.class))).thenReturn(Optional.of(publisher));

        Publisher foundPublisher = publisherService.getPublisherById(publisherId);

        verify(publisherRepository, times(1)).findById(publisherId);
        assert foundPublisher != null;
        assert foundPublisher.getId().equals(publisherId);
    }
}
```