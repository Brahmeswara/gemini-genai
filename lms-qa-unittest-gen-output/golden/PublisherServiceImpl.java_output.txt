```java
package com.example.lws.services;

import com.example.lws.models.UserInput;
import com.example.lws.repositories.UserInputRepository;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;

import java.util.Optional;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.*;

public class UserInputServiceTest {

    @Mock
    private UserInputRepository userInputRepository;

    @InjectMocks
    private UserInputService userInputService;

    @BeforeEach
    public void setUp() {
        MockitoAnnotations.openMocks(this);
    }

    @Test
    public void testSaveUserInput() {
        UserInput userInput = UserInput.builder().input("test").build();

        when(userInputRepository.save(any())).thenReturn(userInput);

        UserInput savedUserInput = userInputService.saveUserInput(userInput);

        assertEquals(userInput.getInput(), savedUserInput.getInput());
        verify(userInputRepository, times(1)).save(any());
    }

    @Test
    public void testUpdateUserInput() {
        UserInput userInput = UserInput.builder().id(1L).input("test").build();

        when(userInputRepository.save(any())).thenReturn(userInput);

        UserInput savedUserInput = userInputService.updateUserInput(userInput);

        assertEquals(userInput.getId(), savedUserInput.getId());
        assertEquals(userInput.getInput(), savedUserInput.getInput());
        verify(userInputRepository, times(1)).save(any());
    }

    @Test
    public void testDeleteUserInput() {
        userInputService.deleteUserInput(1L);

        verify(userInputRepository, times(1)).deleteById(1L);
    }

    @Test
    public void testGetUserInputById() {
        UserInput userInput = UserInput.builder().id(1L).input("test").build();

        when(userInputRepository.findById(1L)).thenReturn(Optional.of(userInput));

        UserInput foundUserInput = userInputService.getUserInputById(1L);

        assertEquals(userInput.getId(), foundUserInput.getId());
        assertEquals(userInput.getInput(), foundUserInput.getInput());
        verify(userInputRepository, times(1)).findById(1L);
    }

    @Test
    public void testGetUserInputById_whenUserInputNotFound() {
        when(userInputRepository.findById(1L)).thenReturn(Optional.empty());

        assertThrows(ResourceNotFoundException.class, () -> userInputService.getUserInputById(1L));
        verify(userInputRepository, times(1)).findById(1L);
    }
}
```